### DISCUSSION TITLE
Connect ESP8266 to a TFT display via SPI

### DISCUSSION THREAD

**Discussion by Santaionut** [Bodmer/TFT_eSPI#2408] [General]
Hi, 

I have next problem, I want to connect a display to ESP8266 via SPI .  The display not have CS pin have only next pins GND, VCC, CLK, MOSI, RES, DC, BLK, MISO. The driver for this display is ILI9341. 

I tried next things:

1) I changed the next lines in User_Setup.h
`#define TFT_SPI_OVERLAP

#define TFT_MISO 12
#define TFT_MOSI 13
#define TFT_SCLK 14`

2) I connected the display to ESP 

ESP           =>  TFT
GPIO12      --      MISO
GPIO13       --     MOSI
GPIO14        --    CLK
3v             --       VCC
GND           --     GND

3) I uploaded on ESP next code 

`#include <TFT_eSPI.h> // Graphics and font library for ILI9341 driver chip
#include <SPI.h>

#define TFT_GREY 0x5AEB // New colour

TFT_eSPI tft = TFT_eSPI();  // Invoke library

void setup(void) {
  Serial.begin(9600);
  tft.init();
  tft.setRotation(2);
}

void loop() {
  
  // Fill screen with grey so we can see the effect of printing with and without 
  // a background colour defined
  tft.fillScreen(TFT_GREY);
  Serial.println("Test ......");
  // Set "cursor" at top left corner of display (0,0) and select font 2
  // (cursor will move to next line automatically during printing with 'tft.println'
  //  or stay on the line is there is room for the text with tft.print)
  tft.setCursor(0, 0, 2);
  // Set the font colour to be white with a black background, set text size multiplier to 1
  tft.setTextColor(TFT_WHITE,TFT_BLACK);  tft.setTextSize(1);
  // We can now plot text on screen using the "print" class
  tft.println("Hello World!");
  
  delay(10000);
}`

And after upload the display not show anything.  

can you help me with this problem ?

Thanks!

**Comment by Bodmer** [comment]
These are the recommended connections:
https://github.com/Bodmer/TFT_eSPI/blob/master/User_Setup.h#L139-L147

If you use these with an unchanged copy of TFT_eSPI then it should work fine.

**Reply by Bodmer** [reply]
Here are the GPIO to PIN_Dxx corresponing signals if your anre not using a NodeMCU board.:
https://github.com/Bodmer/TFT_eSPI/blob/master/User_Setup.h#L139-L147

**Reply by Santaionut** [reply]
Thanks, I'll try and let you know if works well!
